### PR TITLE
Add backticks and crossreferences to parallel/event docs

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: base/workerpool.jl [hunk 1]
```diff
@@ -232,12 +232,14 @@ serialize(s::AbstractSerializer, cp::CachingPool) = throw(ErrorException("Cachin
...
     CachingPool(workers::Vector{Int})
 
-An implementation of an `AbstractWorkerPool`. `remote`, `remotecall_fetch`, `pmap` and other
-remote calls which execute functions remotely, benefit from caching the serialized/deserialized
-functions on the worker nodes, especially for closures which capture large amounts of data.
+An implementation of an `AbstractWorkerPool`.
+[`remote`](:func:`remote`), [`remotecall_fetch`](:func:`remotecall_fetch`),
+[`pmap`](:func:`pmap`) (and other remote calls which execute functions remotely)
+benefit from caching the serialized/deserialized functions on the worker nodes,
+especially closures (which capture large amounts of data).
```
Comment (tkelman) [inline base/workerpool.jl]: not all closures capture large amounts of data

File: base/event.jl [hunk 2]
```diff
@@ -5,8 +5,8 @@
 """
     Condition()
 
-Create an edge-triggered event source that tasks can wait for. Tasks that call `wait` on a
-`Condition` are suspended and queued. Tasks are woken up when `notify` is later called on
+Create an edge-triggered event source that tasks can wait for. Tasks that call [`wait`](:func:`wait`) on a
+`Condition` are suspended and queued. Tasks are woken up when [`notify`](:func:`notify`) is later called on
```
Comment (tkelman) [inline base/event.jl]: there's another `notify` on the next line

File: base/workerpool.jl (hunk 1 above)
Comment (kshyatt) [inline base/workerpool.jl]: "which may capture", then?

Comment (oxinabox) [inline base/workerpool.jl]: "which often capture" ?

Comment (tkelman) [inline base/workerpool.jl]: the benefit of caching is most significant when the amount of captured data is large, but that's not necessarily very common

Comment (kshyatt) [inline base/workerpool.jl]: "may" it is!

**Pull request merged**